app/server/src/configApp.ts

```diff
@@ -0,0 +1,66 @@
+import express from "express";
+import cors from "cors";
+import morgan from "morgan";
+import passport from 'passport';
+import session from 'express-session';
+import PassportGoogle from 'passport-google-oauth20';
+import PassportGithub from 'passport-github';
+import MockStrategy from 'passport-mock-strategy';
+
+import config from './resources/config.json';
+
+export const configApp = (app => {
+    //set up passport
+    passport.use(
+        new PassportGoogle.Strategy(
+            {
+                clientID: config.GOOGLE_CLIENT_ID,
+                clientSecret: config.GOOGLE_CLIENT_SECRET,
+                callbackURL: '/return/google'
+            },
+            (accessToken, refreshToken, profile, cb) => {
+                return cb(null, profile);
+            }
+        )
+    );
+
+    passport.use(
+        new PassportGithub.Strategy({
+            clientID: config.GITHUB_CLIENT_ID,
+            clientSecret: config.GITHUB_CLIENT_SECRET,
+            callbackURL: '/return/github'
+        },
+            (accessToken, refreshToken, profile, cb) => {
+                return cb(null, profile);
+            }
+        )
+    );
+
+    passport.use(new MockStrategy());
```

Comment:
Is there any way to move this out of here? It's not ideal to have test code in the main class like this, but not sure if there's any way to apply passport strategies from the test code. We use Hoverfly in OrderlyWeb to mock Oauth responses from github, some approach like that might be possible?

`configApp` should certainly be unit-testable by mocking passport itself, but integration tests are trickier.